mlrun/api/utils/projects/follower.py

```diff
@@ -122,9 +122,12 @@ def create_project(
                 # for further read: https://docs-sqlalchemy.readthedocs.io/ko/latest/faq/sessions.html
                 # https://docs-sqlalchemy.readthedocs.io/ko/latest/dialects/mysql.html#transaction-isolation-level
                 # https://dev.mysql.com/doc/refman/8.0/en/innodb-transaction-isolation-levels.html
-                # TODO: there are multiple isolation level we can choose, READ COMMITTED seems to solve our issue
-                #  but will require deeper investigation and more test coverage
-                if commit_before_get:
+                # TODO: remove this condition in (1.4.0), if we feel we're stable with the
+                #  new isolation level(READ COMMITTED)
+                if (
+                    commit_before_get
+                    and mlrun.mlconf.db.isolation_level != "READ COMMITTED"
+                ):
```

Comment:
why `and`? I mean why its not `or`? maybe I am missing something

Reply:
we still want to stay with the old behavior and only commit if we are coming from a specific flow which we know runs `GET` twice, but now we will do it only if we are not in `READ COMMITTED` mode.

Reply:
as stated in TODO, this is mainly for backup in case we find issues working with `READ COMMITTED` which requires us to roll back